### DISCUSSION TITLE
How to access the Proxy Contracts?

### DISCUSSION THREAD

**Discussion by denniarems** [wevm/viem#692] [Question]
In I didn't see any examples of how to read and write proxy contract functions in the migration guide. It will be extremely helpful to share that one as well.

**Comment by tmm** [comment]
Not sure what set up you are working with, but should be as easy as using the implementation ABI with the proxy address in any of the functions (e.g. [readContract](https://viem.sh/docs/contract/readContract.html) or [writeContract](https://viem.sh/docs/contract/writeContract.html)).

**Reply by navillanueva** [reply]
do you know if it should be the same using wagmi? I am trying with useContractWrite (wagmi's rapper for writeContract) and passing it the proxy address together with the implementation ABI and the function name in that ABI but still doesn't work

**Reply by tmm** [reply]
Yup, it should. Best to keep discussions about wagmi over there.
